### ISSUE TITLE
error 12501

### ISSUE THREAD

**Issue by isunnynegi** [EddyVerbruggen/cordova-plugin-googleplus#307]
**by this i have got error code 12501 everytimes**

> window.plugins.googleplus.login({
				'scopes': 'profile email',
				'webClientId': '803433772325-ptuf8dssbmgh500bgkube2srdm10utve.apps.googleusercontent.com'
			},
			function (obj) {
				alert('response');
				alert(JSON.stringify(obj)); // do something useful instead of alerting
				/* GoogleName = googleUser.getBasicProfile().getName();
				GoogleId = googleUser.getBasicProfile().getId();
				ImgUser = googleUser.getBasicProfile().getImageUrl()+'?sz=250';
				GoogleEmail = googleUser.getBasicProfile().getEmail();
				SignInDirect('google'); */
			},function (msg) {
				alert('error');
				alert(JSON.stringify(msg));
			}
		);

[http://stackoverflow.com/questions/40677871/cordova-plugin-googleplus-gives-error-12501](url)

**Comment by signalpoint** [comment]
@isunnynegi The following issue helped me resolve the 12501 error: https://github.com/EddyVerbruggen/cordova-plugin-googleplus/issues/165

When I generated my keys using this form https://developers.google.com/mobile/add?platform=android&cntapi=signin (as mentioned in the README), I ended up using "Android key" in my app, which is incorrect. Once I used the "Web client" id, it worked.

This particular comment explains why we use the web client id instead of the Android id: https://github.com/EddyVerbruggen/cordova-plugin-googleplus/issues/165#issuecomment-205941390

**Comment by omhybrid** [comment]
i have same error

**Comment by st1s** [comment]
same

**Comment by Ashirogi-Muto** [comment]
I am also getting the same error even after using "Web Application ID" for webClientId.
My code:-
```
this.platform.ready().then(() => {
      GooglePlus.login({
        'webClientId': 'xxxxxxxxxxxxxxxxxx.apps.googleusercontent.com',
        'offline': true
      }).then(function (repsonse) {
        console.log("the response " + repsonse);
      }).catch(function (errorResp) {
        console.log("error " + errorResp);
      });
    });
```
I get the options to choose the account for login in my device but after selecting an account I get error 12501

**Comment by sammuggPS** [comment]
Hi @Ashirogi-Muto and Company,

I believe 12501 is more commonly caused by an incorrect SHA1 being used to set up your project with Google. Make sure that the SHA1 of the build you are testing matches what you used in the Developer's Console.

Cheers.

**Comment by Ashirogi-Muto** [comment]
Thanks. I'll surely once again check the SHA1 for my project.

**Comment by Ashirogi-Muto** [comment]
Hey @sammuggPS  can you help me with the SHA1. I found nothing on my developers console regarding this issue

**Comment by sammuggPS** [comment]
SHA1 is part of your configuration for Android devices. So on your project API credentials page, find the configuration you have set up for Android and cross-reference the SHA1 there with the one you're using in your development environment.

**Comment by Ashirogi-Muto** [comment]
I have only generated  web application API for my ionic 2 project. I haven't configured anything for an Android device

**Comment by Ashirogi-Muto** [comment]
Do I have to generate the SHA1 fingerprint to test my app on a real Android device?

**Comment by Ashirogi-Muto** [comment]
@sammuggPS  Never mind. I read some more closed issues and figured it out. Thanks for your help.

**Comment by punjasin** [comment]
i fix it u have to put client id that u generate in the config.xml and then
at this part
 GooglePlus.login({
        'webClientId': 'xxxxxxxxxxxxxxxxxx.apps.googleusercontent.com',

u have to generate again using javascript type not android it will work
or u delete this line it will work too but u will not get authkey

**Comment by tesnep** [comment]
@Ashirogi-Muto Do you mind sharing what you ended up doing to get it working?

**Comment by Ashirogi-Muto** [comment]
I was getting the error while testing it on my android device. So as suggested by @sammuggPS  I generated an SHA1 key for my debugging APK and it worked properly then. There was no problem in the code.

**Comment by cpup22** [comment]
I kept getting this error and was able to get it to go away by removing the cordova plugin and adding again.  *magic* :)

**Comment by Barathwaja** [comment]
i was able to get rid of error 12501.. please give the SHA-1 key correctly ...

1 : create an ionic proj.
2 : after that add the plugin and stuffs given by Eddy
3 : then create android platform and add keystore (if release build) or debug.keystore available in this path %USERPROFILE%/.android/debug.keystore.
4 : google how to find SHA-1 in in debug mode and copy and paste in reqd. proj
5 : finally build apk and it wll work,..

**Comment by vshanbha** [comment]
I am having a hard time with 12501. I am working with Ionic 1 + Firebase. I followed the instructions / demo to set up the authentication and when I click "google sign in" button on my app, Googles sign in dialogue runs on the emulator and shows me list of accounts. However after I select an account my app logs above error code in console. What am I missing ?

Debugging efforts - 

1. webclient ID  
I checked that 
'webClientId': 'xxxxxxxxxxxxxxxxxx.apps.googleusercontent.com' is added in appropriate place in my javascript code.

2.  Reverseclient id
config.xml contains 'com.googleusercontent.apps.xxxxxxxxxxxxxxxxx' inside plugin configuration

3. SHA-1 
I have created an "Android" project in firebase console and I have added SHA-1 signature from my debug keystore inside the Android project config from firebase. The plugin docs for this project say _"There is no need to add the generated google-services.json file into your cordova project."_

**Comment by J-Yen** [comment]
I fixed the problem by specifying the location of the keystore:
https://www.webniraj.com/2016/11/21/cordova-changing-the-default-debug-keystore-for-android-applications/

**Comment by schmorrison** [comment]
I encountered this problem while trying to build an Ionic2 app. Although the instructions below have been provided a whole bunch, I figure I would rewrite them for chronological relevance.

Google login worked fine in the browser but threw the `12501` error when trying to `GooglePlus.login()` on an android device. It turns out this was because the tutorial I followed suggested to add the ClientID from the google-services.json to the REVERSED_CLIENT_ID value when installing the plugin and in the WebClientID parameter in `GooglePlus.login()` function.

I ended up going to [API Credential Console](https://console.developers.google.com) and navigating to the project(It should be autocreated by Firebase in the correct Users project list), and copying the projects "WebClientID" (which should also be autocreated by Firebase) into the `GooglePlus.login()` function.

Works fine now.

Thanks everyone
